test/unit/processing/components/get-schedule-dates.test.js

```diff
@@ -93,7 +93,12 @@ describe('get schedule dates', () => {
 
   test('should insert top up adjustment after last outstanding payment in original schedule', () => {
```

Comment:
What if `deltaValue` is 0?

I don't believe we'd get the correct behaviour -- one for a tech debt ticket.

I appreciate, perhaps we should never get a 0 delta here or it's v difficult to but feel like this case is not covered -- outside scope of this ticket however

Reply:
when 0 (no adjustment), that means there is no top-up or  reduction, hence you need do nothing. Also, this is actually existing functionality which the current ticket does not cover.